logical/logical.go

```diff
@@ -47,6 +64,17 @@ type Backend interface {
 	// to the backend. The backend can use this to clear any caches or reset
 	// internal state as needed.
 	InvalidateKey(key string)
+
+	// Configure is used to set up the backend based on the provided backend
+	// configuration. This is primarily used in the the context of backend
+	// plugins, where the RPC server has no context of the core's config.
+	// This is the equivalent of calling framework.Backend.Setup(conf), but
+	// from the RPC client to configure components such as the logger, storage,
+	// system view, etc.
+	Configure(*BackendConfig) error
```

Comment:
Sure, but that's a compile time catch so between that and `ag` should be pretty easy.

Also you won't be able to change it to that since the changed signature means that `logical.Backend` is not part of what's returned.

The other option is just keep the existing Setup function signature from framework (which has `logical.Backend, error`). Then you don't have to change any existing backends.